src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
```

Comment:
I'd probably exit early if the search term is empty.

Reply:
```js
const regex = /^[a-zA-Z0-9_]+$/
    try {
      const url =
        regex.test(this.state.searchTerm)
          ? `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`
          : `https://icanhazdadjoke.com/search`;
      const fetchUrl = fetch(url, {
```

So I added this, which works and stops any empty strings being added...
I wanted to add style to the input to show there is an error, perhaps adding a value to state like `{valid: true}` which is updated if the regex is valid or not. 

Would that be the right path here?